packages/edit-site/src/hooks/commands/use-edit-mode-commands.js

```diff
@@ -295,6 +286,9 @@ function useEditUICommands() {
 		label: __( 'Toggle top toolbar' ),
 		callback: ( { close } ) => {
 			toggle( 'core/edit-site', 'fixedToolbar' );
+			if ( isDistractionFree ) {
```

Comment:
For now this is the only one left from the issue, because we call directly the `preferences` store action.